controller/api/models.py

```diff
@@ -292,6 +292,7 @@ def _start_containers(self, to_add):
         if any(c.state != 'created' for c in to_add):
             err = 'aborting, failed to create some containers'
             log_event(self, err, logging.ERROR)
+            self._destroy_containers(to_add)
```

Comment:
I think this should be part of .deploy() and .scale() instead. If I'm following the code correctly, what'll happen is that the new containers will be destroyed *as well as* the old containers. In other words, we should be catching the RuntimeError in .scale() and in .deploy(). If an error occurs while running _start_containers() we can destroy that list of containers, prevent .scale() or .deploy() from destroying the old containers and then re-raise the original exception to ensure that other parts of the system still work as intended. How does that sound?